packages/alpinejs/src/directives/x-if.js

```diff
@@ -22,7 +24,18 @@ directive('if', (el, { expression }, { effect, cleanup }) => {
 
         el._x_currentIfEl = clone
 
-        el._x_undoIf = () => { clone.remove(); delete el._x_currentIfEl }
+        el._x_undoIf = () => {
+            walk(clone, (node) => {
+                if (!!node._x_effects) {
+                    node._x_effects.forEach((effect) => {
+                        dequeueJob(effect)
+                    })
```

Comment:
Could just call `node._x_effects.forEach(dequeueJob)` without the arrow function.